implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java

```diff
@@ -314,7 +329,33 @@ boolean isAddDiscoveredInterceptors() {
 
     @Override
     public SmallRyeConfig build() {
-        return new SmallRyeConfig(this);
+        ConfigMappingProvider mappingProvider = mappingsBuilder.build();
+        if (!defaultValues.isEmpty() || !mappingProvider.getDefaultValues().isEmpty()) {
+            final KeyMap<String> mappingProviderDefaultValues = mappingProvider.getDefaultValues();
+            defaultValues.forEach((key, value) -> mappingProviderDefaultValues.findOrAdd(key).putRootValue(value));
+            withSources(new KeyMapBackedConfigSource("DefaultValuesConfigSource", mappingProviderDefaultValues) {
+                private static final long serialVersionUID = 7847969724478280439L;
+
+                @Override
+                public Map<String, String> getProperties() {
+                    return new HashMap<>();
```

Comment:
Use `Collections.emptyMap()` instead of constructing a new instance every time the method is called.